Add "path" key to Compass.sublime-build

graphical-iain opened the pull request:
Added path key to account for applications running at a different path in El Capitan.

Before I added in the path key, I would get `[ERROR] compass not found. Make sure it exists in your PATH.` This is because the path was only /usr/bin, but compass existed at /user/local/bin.

I'm not sure if an override is necessary for windows, and if it is, I don't know what it should be.

graphical-iain committed:
Add "path" key to Compass.sublime-build

Added path key to account for applications running at a different path
on El Capitan.

renestalder commented:
I'm not quite sure about this. Sublime Text looks in the PATH variable for the executable defined in the cmd of the build config. This way we can get sure the user of the Compass build system is able to use different environment like rvm.

IMHO changing this to static strings would override the PATH variable, thus making it a must to install Compass at the given path of the build system.

You couldn't fix your error with setting the correct location of Compass via PATH?

graphical-iain commented:
I didn't try updating my PATH, but I assume that would work. However, I only started receiving the error after I updated to El Capitan, so I figured others would encounter the same error and it would be smart to put the default PATH for compass into the build system, that way it wouldn't be necessary for everyone to update their PATH.
I understand the hesitancy to hard code the path, so perhaps I'll play with it a bit more to see if I can get it to work without hard coding a path. 
Have you heard from anyone else that they're having issues on El Capitan?

renestalder commented:
I only heard from the error, but not El Capitan specific. Mostly it came from people who didn't set the path correctly and use other environments, as mentioned, or do not setup their project as described in the readme.

I guess the better fix would be setting the path per user on the system to make sure, it's set for all applications. And let the application take the PATH variable.

I will wait with the merge because at the moment it could break many installations.
Also the code only should address OSX. That should be possible with platform specific rules, if this part of the documentation is still correct (couldn't find it in newer documentations).
http://sublimetext.info/docs/en/reference/build_systems.html

graphical-iain committed:
Move path to OSX specific

graphical-iain committed:
Update path

Add $PATH to the end of the updated path string to preserve the user set path, for instances where the user may have a custom path set.

graphical-iain commented:
@renestalder Some of my coworkers encountered the same issue today, and I discovered that I can preserve the user set path by adding the $PATH variable. So I can add the custom path, while preserving ones that might have been custom set by the user.

renestalder commented:
Thank you. I will look add this in the next couple of days. Did you or your coworkers test your solution on a clean installation?
